apps/remix-ide/src/app/files/remixd-handle.js

```diff
@@ -95,6 +95,7 @@ export class RemixdHandle extends WebsocketPlugin {
           label: 'Connect',
           fn: () => {
             try {
+              this.locahostProvider.event.trigger('loading')
```

Comment:
normally `event` should be triggered by the class, not from an exernal one.
At least could you have a `preInit()` function in `locahostProvider` and call that one?

Reply:
Okay